pkg/reconciler/taskrun/taskrun.go

```diff
@@ -300,6 +300,12 @@ func (c *Reconciler) prepare(ctx context.Context, tr *v1beta1.TaskRun) (*v1beta1
 		return nil, nil, controller.NewPermanentError(err)
 	}
 
+	if err := validateContextVariables(taskSpec, tr); err != nil {
```

Comment:
I am having trouble seeing this one as non-exported in `taskrun` and the other one *exported* in `pipelinerun/resources`. Could they share code even ? (because we are "validating" the same thing, be it taskrun or other objects)